ert3/storage/_storage.py

```diff
@@ -3,29 +3,98 @@
 from pathlib import Path
 from typing import Any, Dict, Iterable, Optional, cast, Set
 import io
-
+import logging
 import cloudpickle
 import requests
 import ert3
 
+from ert_shared.storage.connection import get_info
+
+logger = logging.getLogger(__file__)
 
-_STORAGE_URL = "http://localhost:8000"
-_DATA = "__data__"
-_PARAMETERS = "__parameters__"
+_STORAGE_TOKEN = None
+_STORAGE_URL = None
 _ENSEMBLE_RECORDS = "__ensemble_records__"
 _SPECIAL_KEYS = (_ENSEMBLE_RECORDS,)
 
 
+def _get_from_server(
+    url: str,
+    headers: Optional[Dict[Any, Any]] = None,
+    status_code: int = 200,
+    **kwargs: Dict[Any, Any],
+) -> requests.Response:
+    global _STORAGE_URL, _STORAGE_TOKEN  # pylint: disable=global-statement
+
+    if _STORAGE_URL is None:
+        info = get_info()
+        _STORAGE_URL = info["baseurl"]
+        _STORAGE_TOKEN = info["auth"][1]
+    if not headers:
+        headers = {}
+    headers["Token"] = _STORAGE_TOKEN
+
+    resp = requests.get(f"{_STORAGE_URL}/{url}", headers=headers, **kwargs)
+    if resp.status_code != status_code:
+        logger.error("Failed to fetch from %s. Response: %s", url, resp.text)
+
+    return resp
+
+
+def _delete_on_server(
+    url: str, headers: Optional[Dict[Any, Any]] = None, status_code: int = 200
+) -> requests.Response:
+    global _STORAGE_URL, _STORAGE_TOKEN  # pylint: disable=global-statement
+    if _STORAGE_URL is None:
+        info = get_info()
+        _STORAGE_URL = info["baseurl"]
+        _STORAGE_TOKEN = info["auth"][1]
+
+    if not headers:
+        headers = {}
+    headers["Token"] = _STORAGE_TOKEN
+    resp = requests.delete(
+        f"{_STORAGE_URL}/{url}",
+        headers=headers,
+    )
+    if resp.status_code != status_code:
+        logger.error("Failed to delete %s. Response: %s", url, resp.text)
+
+    return resp
+
+
+def _post_to_server(
+    url: str,
+    headers: Optional[Dict[Any, Any]] = None,
+    status_code: int = 200,
+    **kwargs: Dict[Any, Any],
+) -> requests.Response:
+    global _STORAGE_URL, _STORAGE_TOKEN  # pylint: disable=global-statement
```

Comment:
Lots of logic here that is duplicated in the other functions in this module. Move setting of `_STORAGE_URL`and `_STORAGE_TOKEN` into its own function, eg `_assert_server_info`.